addons/azureSets/azuresets.lua

```diff
@@ -68,19 +68,36 @@ windower.register_event('login', initialize)
 
 windower.register_event('job change', initialize:cond(function(job) return job == 16 end))
 
-function set_spells(spellset)
+function set_spells(spellset, setmode)
     if windower.ffxi.get_player()['main_job_id'] ~= 16 --[[and windower.ffxi.get_player()['sub_job_id'] ~= 16]] then return nil end
     if settings.spellsets[spellset] == nil then return end
-    if settings.spellsets[spellset]:equals(get_current_spellset()) then
+    if is_spellset_equipped(settings.spellsets[spellset]) then
         error(spellset..' was already equipped.')
         return
     end
 
     log('Starting to set '..spellset..'.')
-    set_spells_from_spellset(spellset, 'remove')
+    if setmode=='clearfirst' or (setmode == nil and settings.setmode == 'clearfirst') then
+        remove_all_spells(nil)
+        windower.send_command('@wait .65;lua i azuresets set_spells_from_spellset '..spellset..' add')
+    elseif setmode == 'preservetraits' or (setmode == nil and settings.setmode == 'preservetraits') or settings.setmode == nil then
+        set_spells_from_spellset(spellset, 'remove')
+    end
     return
 end
 
+function is_spellset_equipped(spellset)
+    if spellset:equals(get_current_spellset()) then return true end
+    local currentSet = get_current_spellset()
+    for k,v in pairs(spellset) do
+        if not currentSet:contains(v:lower()) then return false end
+    end
+    for k,v in pairs(currentSet) do
+        if not spellset:contains(v:lower()) then return false end
+    end
+    return true
+end
```

Comment:
That works. Like I said above, I'm more of a fan of table validation. To be honest, these tables should be using IDs, not names, but that's too much of a refactoring for this pull request. Anyway, the mapping works, but it can be simplified:
```lua
function is_spellset_equipped(spellset)
    return S(spellset):map(string.lower) == S(get_current_spellset())
end
```